cpp/CMakeLists.txt

```diff
@@ -83,25 +86,54 @@ function (find_required_program NAME FILENAME DESCRIPTION)
 endfunction (find_required_program)
 
 # Options that can be passed to CMake using 'cmake -DKEY=VALUE'.
-option ("BUILD_GEOCODER" "Build the offline phone number geocoder" "ON")
-option ("REGENERATE_METADATA" "Regenerate metadata instead of using it from the source tree" "ON")
-option ("USE_ALTERNATE_FORMATS" "Use alternate formats" "ON")
-option ("USE_PROTOBUF_LITE" "Link to protobuf-lite" "OFF")
-option ("USE_BOOST" "Use Boost" "ON")
-option ("USE_ICU_REGEXP" "Use ICU regexp engine" "ON")
-option ("USE_LITE_METADATA" "Use lite metadata" "OFF")
-option ("USE_RE2" "Use RE2" "OFF")
-option ("USE_STD_MAP" "Force the use of std::map" "OFF")
-option ("BUILD_STATIC_LIB" "Build static libraries" "ON")
-option ("USE_STDMUTEX" "Use C++ 2011 std::mutex for multi-threading" "OFF")
-option ("USE_POSIX_THREAD" "Use Posix api for multi-threading" "OFF")
-
-if (${USE_ALTERNATE_FORMATS} STREQUAL "ON")
+option (BUILD_GEOCODER "Build the offline phone number geocoder" ON)
+option (REGENERATE_METADATA "Regenerate metadata instead of using it from the source tree" ON)
+option (USE_ALTERNATE_FORMATS "Use alternate formats" ON)
+option (USE_PROTOBUF_LITE "Link to protobuf-lite" OFF)
+option (USE_BOOST "Use Boost" ON)
+option (USE_ICU_REGEXP "Use ICU regexp engine" ON)
+option (USE_LITE_METADATA "Use lite metadata" OFF)
+option (USE_RE2 "Use RE2" OFF)
+option (USE_STD_MAP "Force the use of std::map" OFF)
+option (BUILD_STATIC_LIB "Build static libraries" ON)
```

Comment:
About these new options, could you please add some notes in cpp/README file.
It helps readers to know about these options.